packages/code-du-travail-frontend/src/lib/decision/walker.js

```diff
@@ -0,0 +1,154 @@
+//@flow
+
+/*
+
+
+*/
+
+//type Filters = Object;
+
+const sum = array => array.reduce((a, c) => a + c, 0);
+
+const mean = array => sum(array) / array.length;
+
+const getVariance = array =>
+  mean(array.map(num => Math.pow(num - mean(array), 2)));
+
+/*
+type TagValueCount = Number;
+type TagValues = {
+  [key: string]: TagValueCount
+};
+
+type TagValue = String | String[];
+type FilterValue = String;
+
+type TagsCount = { [key: string]: TagValues };
+
+type Doc = {
+  id: String,
+  tags: {
+    [key: string]: TagValue
+  }
+};
+
+type Docs = Doc[];
+*/
+
+// compute number of docs for each tag+value pair
+export const getDocsCountByTag = ({ docs, tags }) => {
+  const tagsCount = {} /*: TagsCount*/;
+  docs.filter(doc => doc.tags).forEach(doc => {
+    Object.keys(doc.tags)
+      .filter(tag => (tags && tags.length ? tags.indexOf(tag) > -1 : tag))
+      .forEach(docTag => {
+        const increment = docTagValue => {
+          if (tagsCount[docTag] === undefined) {
+            tagsCount[docTag] = {};
+          }
+          if (!docTagValue) {
+            return;
+          }
+          if (tagsCount[docTag][docTagValue] === undefined) {
+            tagsCount[docTag][docTagValue] = 0;
+          }
+          tagsCount[docTag][docTagValue]++;
+        };
+        if (Array.isArray(doc.tags[docTag])) {
+          doc.tags[docTag].forEach(value => {
+            increment(value);
+          });
+        } else {
+          increment(doc.tags[docTag]);
+        }
+      });
+  });
+  return tagsCount;
+};
+
+// add tag details so we can make the best match later on
+export const getTagData = (tag, values) => {
+  const allValues = Object.values(values),
+    documents = sum(allValues),
+    variance = getVariance(allValues),
+    count = Object.keys(values).length;
+
+  // TODO this is where we should compute something better, ex: popularity
```

Comment:
Can you please add your user handle on this as TODO comments are hard to track and blame.

Similar shape as than https://www.conventionalcommits.org 

```js
// TODO(douglasduteil): title
// Your full message
// Refs
```